src/js/elements/subindicator_filter/filter_row.js

```diff
@@ -0,0 +1,231 @@
+import {Dropdown, DropdownModel} from "./dropdown";
+import {Observable} from "../../utils";
+
+/**
+ *
+ */
+class FilterRowModel extends Observable {
+    static EVENTS = {
+        'updated': 'filterRowModel.updated',  // triggered when new datafiltermodel is set
+        'indicatorSelected': 'filterRowModel.indicatorSelected' //
+    }
+
+    static ALL_VALUES = 'All values';
+
+    constructor(dataFilterModel, isDefault, isExtra, defaultIndicatorText = 'All indicators', defaultSubindicatorText = 'All values') {
+        super()
+        this._isDefault = isDefault;
+        this._isExtra = isExtra;
+        this._currentIndicatorValue = null;
+        this._currentSubindicatorValue = null;
+        this._defaultIndicatorText = defaultIndicatorText;
+        this._defaultSubindicatorText = defaultSubindicatorText
+        this._dataFilterModel = null;
+
+        this.setDataFilterModel(dataFilterModel)
+    }
+
+    get dataFilterModel() {
+        return this._dataFilterModel;
+    }
+
+    get indicatorValues() {
+        /**
+         * Returns the available indicatorValues or [] if none are available
+         */
+        if (this.dataFilterModel != null)
+            return [FilterRowModel.ALL_VALUES].concat(this.dataFilterModel.availableFilterNames);
+        return []
+    }
+
+    get subindicatorValues() {
+        /**
+         * Returns the subindicator values for the currently selected indicatorValue. Returns [] if this is not possible
+         */
+        if (this.dataFilterModel != null) {
+            if (this._currentIndicatorValue != null && this._dataFilterModel.groupLookup[this._currentIndicatorValue] != undefined)
+                return this._dataFilterModel.groupLookup[this._currentIndicatorValue].values;
```

Comment:
I think change `this._dataFilterModel` to `this.dataFilterModel` so that we use the getter instead of accessing it directly